internal/cache/backend.go

```diff
@@ -43,14 +43,27 @@ func (b *Backend) Remove(ctx context.Context, h restic.Handle) error {
 	return b.Cache.Remove(h)
 }
 
-var autoCacheTypes = map[restic.FileType]struct{}{
-	restic.IndexFile:    {},
-	restic.SnapshotFile: {},
+func autoCacheTypes(ft restic.FileType, bt restic.BlobType) bool {
+	if ft == restic.IndexFile || ft == restic.SnapshotFile ||
+		(ft == restic.DataFile && bt != restic.DataBlob) {
+		return true
+	}
+	return false
```

Comment:
```
return ft == restic.IndexFile || ft == restic.SnapshotFile ||
	ft == restic.DataFile && bt != restic.DataBlob
```

or

```
switch ft {
case restic.IndexFile, restic.SnapshotFile:
	return true
case restic.DataFile:
	return bt != restic.DataBlob
}
```

Also, this conflicts with #2838, but the changes are very similar.

Reply:
is changed.